### DISCUSSION TITLE
The "examples" fields do not render in swagger-ui 5 as the documentation suggests.

### DISCUSSION THREAD

**Discussion by UltimateLobster** [fastapi/fastapi#9839] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

import uvicorn
from fastapi import Body, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None


@app.put("/items/{item_id}")
async def update_item(
    *,
    item_id: int,
    item: Annotated[
        Item,
        Body(
            examples=[
                {
                    "name": "Foo",
                    "description": "A very nice Item",
                    "price": 35.4,
                    "tax": 3.2,
                },
                {
                    "name": "Bar",
                    "price": "35.4",
                },
                {
                    "name": "Baz",
                    "price": "thirty five point four",
                },
            ],
        ),
    ],
):
    results = {"item_id": item_id, "item": item}
    return results

uvicorn.run(app)
```

### Description

First of all, I LOVE fastapi. This package is amazing and all of my co-workers enjoy using it :D

The [fastapi documentation](https://fastapi.tiangolo.com/tutorial/schema-extra-example/?h=examples#example-in-the-docs-ui), suggests that the new swagger allows us to see every example we specified. However, when I test it using the provided example code, I only see the first of the examples (The way the old swagger UI did). 

Is there any additional thing I need to do? If so, I would love to add it to the documentation. 

### Operating System

Windows

### FastAPI Version

0.100.0

### Python Version

3.11.4

### Additional Context

![image](https://github.com/tiangolo/fastapi/assets/21122724/7abc6aa4-bd28-4818-b2a8-ca4159f5bc65)

**Comment by mikeedjones** [comment]
Interesting that in the [example with multiple examples](https://fastapi.tiangolo.com/tutorial/schema-extra-example/?h=examples#examples-in-the-docs-ui) each example has a name in the dropdown - not clear where that has been set? 

The [OpenAPI schema](https://spec.openapis.org/oas/v3.1.0#pathsObject) defines the examples as a Map - which isn't how its implemented here.


EDIT: 
Investigating further I don't think the specification is implemented correctly here? The examples are ingested as a member of Schema - when in the [pathsObject spec](https://spec.openapis.org/oas/v3.1.0#fixed-fields-9) the are separate members of the pathsObject. 

```python
request_media_content: Dict[str, Any] = {"schema": body_schema}
```
[line 151 here](https://github.com/tiangolo/fastapi/blob/c165be380f87b7b079f98c5134cccef0c9d67e4d/fastapi/openapi/utils.py#L151)

should be more like 

```python
request_media_content: Dict[str, Any] = {"schema": body_schema, "examples": examples}
```

EDIT2:

Discussion here. I think the fastAPI docs are misleading - swagger does not seem to support the JSONSchema examples correctly AFAICT. 

https://github.com/tiangolo/fastapi/discussions/9794

**Comment by lagom0327** [comment]
```python
@app.put("/items/{item_id}")
async def update_item(
    *,
    item_id: int,
    item: Annotated[
        Item,
        Body(
            examples={
                "A normal example": {
                    "value": {
                        "name": "Foo",
                        "description": "A very nice Item",
                        "price": 35.4,
                        "tax": 3.2,
                    }
                },
                "An example with converted data": {
                    "value" : {
                    "name": "Bar",
                    "price": "35.4",
                    }
                },
                "Invalid data is rejected with an error": {
                    "value":{
                    "name": "Baz",
                    "price": "thirty five point four",
                }},
            },
        ),
    ],
):
    results = {"item_id": item_id, "item": item}
    return results
```
![image](https://github.com/tiangolo/fastapi/assets/49493665/f6ddaee7-25e4-4f2f-8fc3-48f58dd6b70a)

It works for me. But I have an older version of fastapi , so I don't know if it will help you.

## FastAPI Version

0.98.0
## Python Version

3.10.6

**Reply by mikeedjones** [reply]
yes, this change came in with 0.99

**Comment by ko330** [comment]
fastapi 0.101.0 also have this problem
I tried 0.98.0 is OK
is it a bug?

**Reply by UltimateLobster** [reply]
The discussion continued [here](https://github.com/tiangolo/fastapi/pull/9866)

Basically, this is a regression but not __necessarily__ a bug. As I understand it, swagger 4 used their own custom "examples" field which was indeed a mapping. In swagger 5, they moved on to use jsonschema's native "examples" field which is a list (and therefore does not support mappings). 
There is a different way to specify multiple examples to swagger, however if we want FastAPI to support them by default, it will require either:
1. Changing the default way FastAPI generates the openapi.json file in a way which one could argue is a patch just for swagger and might break the division of responsibility in the code (or at the very least, a patch that might cause other unexpected behaviour in the final openapi.json which can cause future troubles in backwards compatibility).
2. Adding a entire new way of specifying this mapping through the route definition which could cause code duplication and reduce the beauty and simplicity of the API.

Someone created a PR to go with option 1 but we didn't hear from @tiangolo and don't know if he's willing to accept this PR.

**Comment by tiangolo** [comment, marked as answer]
There's a new parameter `openapi_examples` just for this, to allow declaring examples that show up in Swagger UI while it implements support for multiple JSON Schema examples. It's available now in FastAPI 0.103.0, the new docs are here: https://fastapi.tiangolo.com/tutorial/schema-extra-example/#openapi-specific-examples 🎉
